### PR TITLE
Add sof support for vangogh platform

### PR DESCRIPTION
**Pull request by Venkata-Prasad-Potturu**
There is a dependency with the below patch series.
https://patchwork.kernel.org/project/alsa-devel/list/?series=763439

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Comment by bhiregoudar** [comment]
> Can one of the admins verify this patch?
> 
> > reply **test this please** to run this test once

test this please

**Review by vijendarmukunda — DISMISSED** [review]
patch LGTM

File: sound/soc/sof/amd/acp-loader.c [hunk 1]
```diff
@@ -219,3 +223,45 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
...
+
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: sof signed firmware code bin is missing, you might need to\n");
+		dev_err(sdev->dev,
+			"download it from https://github.com/thesofproject/sof-bin/\n");
+		return ret;
+
+	} else {
+		dev_dbg(sdev->dev, "request_firmware %s successful\n",
```
Comment (vijendarmukunda) [inline sound/soc/sof/amd/acp-loader.c]: it can be restricted to single line.

File: sound/soc/sof/amd/acp-loader.c [hunk 2]
```diff
@@ -219,3 +223,45 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
...
 EXPORT_SYMBOL_NS(acp_sof_dsp_run, SND_SOC_SOF_AMD_COMMON);
+
+int acp_sof_load_signed_firmware(struct snd_sof_dev *sdev)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct acp_dev_data *adata = plat_data->hw_pdata;
+	int ret;
+
+	ret = request_firmware(&sdev->basefw.fw, adata->fw_code_bin, sdev->dev);
+
```
Comment (vijendarmukunda) [inline sound/soc/sof/amd/acp-loader.c]: white line is not required

File: sound/soc/sof/amd/acp-loader.c [hunk 3]
```diff
@@ -219,3 +223,45 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
...
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: sof signed firmware code bin is missing, you might need to\n");
+		dev_err(sdev->dev,
+			"download it from https://github.com/thesofproject/sof-bin/\n");
+		return ret;
+
```
Comment (vijendarmukunda) [inline sound/soc/sof/amd/acp-loader.c]: white line not required

File: sound/soc/sof/amd/acp-loader.c [hunk 4]
```diff
@@ -219,3 +223,45 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
...
+	ret = request_firmware(&adata->fw_dbin, adata->fw_data_bin, sdev->dev);
+
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: sof signed firmware data bin is missing, you might need to\n");
+		dev_err(sdev->dev,
+			"download it from https://github.com/thesofproject/sof-bin/\n");
+		return ret;
+
```
Comment (vijendarmukunda) [inline sound/soc/sof/amd/acp-loader.c:247]: white line is not required

File: sound/soc/sof/amd/acp-loader.c (hunk 1 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp-loader.c]: Done

File: sound/soc/sof/amd/acp-loader.c (hunk 3 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp-loader.c]: Done

File: sound/soc/sof/amd/acp-loader.c (hunk 2 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp-loader.c]: Done

File: sound/soc/sof/amd/acp-loader.c (hunk 4 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp-loader.c:247]: Done

**Comment by Venkata-Prasad-Potturu** [comment]
@plbossart @ranj063 
Could you please review the patches

**Review by vijendarmukunda — DISMISSED** [review]

**Comment by plbossart** [comment]
test this please

File: sound/soc/sof/amd/acp-loader.c [hunk 5]
```diff
@@ -219,3 +223,40 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
...
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct acp_dev_data *adata = plat_data->hw_pdata;
+	int ret;
+
+	ret = request_firmware(&sdev->basefw.fw, adata->fw_code_bin, sdev->dev);
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: sof signed firmware code bin is missing, you might need to\n");
+		dev_err(sdev->dev,
+			"download it from https://github.com/thesofproject/sof-bin/\n");
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-loader.c]: is this correct? I thought AMD did not release firmware through sof-bin?

File: sound/soc/sof/amd/acp.h [hunk 6]
```diff
@@ -82,6 +83,8 @@
 #define SRAM1_SIZE				0x13A000
 #define PROBE_STATUS_BIT			BIT(31)
 
+#define FW_SIGNATURE				0x100
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.h]: prefix? FW_ is awfully generic.

File: sound/soc/sof/amd/acp.c [hunk 7]
```diff
@@ -529,6 +547,19 @@ int amd_sof_acp_probe(struct snd_sof_dev *sdev)
...
 
 	adata->signed_fw_image = false;
+	dmi_id = dmi_first_match(acp_sof_quirk_table);
+	if (dmi_id && dmi_id->driver_data) {
+		adata->fw_code_bin = kasprintf(GFP_KERNEL, "%s/sof-%s-code.bin",
+					       plat_data->fw_filename_prefix,
+					       chip->name);
+		adata->fw_data_bin = kasprintf(GFP_KERNEL, "%s/sof-%s-data.bin",
+					       plat_data->fw_filename_prefix,
+					       chip->name);
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.c:559]: need to test that kasprintf worked....

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/sof/amd/acp-loader.c (hunk 5 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp-loader.c]: Yes, that's not in AMD plan.
Sorry for the inconvenience, I will change the error message.

File: sound/soc/sof/amd/acp.h (hunk 6 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp.h]: Okay, will change it to FIRMWARE_SIGNATURE

File: sound/soc/sof/amd/acp.c (hunk 7 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp.c:559]: Yes, It is working fine, debug log attached below
[    2.923341] snd_sof_amd_vangogh 0000:04:00.5: fw_code_bin:amd/sof/sof-vangogh-code.bin, fw_data_bin:amd/sof/sof-vangogh-data.bin

**Review by vijendarmukunda — DISMISSED** [review]

File: sound/soc/sof/amd/acp.c [hunk 8]
```diff
@@ -235,6 +251,7 @@ int configure_and_run_sha_dma(struct acp_dev_data *adata, void *image_addr,
 		}
 	}
 
+	snd_sof_dsp_write(sdev, ACP_DSP_BAR, ACP_SHA_DMA_INCLUDE_HDR, ACP_SHA_HEADER);
```
Comment (vijendarmukunda) [inline sound/soc/sof/amd/acp.c]: This is required for Signed Firmware only. It will break the Unsigned fw loading. Please add condition check

Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp.c]: Okay agree, Thanks
I will add condition check.

**Review by vijendarmukunda — DISMISSED** [review]

File: sound/soc/sof/amd/pci-vangogh.c [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+	.default_fw_path	= {
+		[SOF_IPC] = "amd/sof",
+	},
+	.default_tplg_path	= {
+		[SOF_IPC] = "amd/sof-tplg",
+	},
+	.default_fw_filename	= {
+		[SOF_IPC] = "sof-vangogh.ri",
+	},
+	.nocodec_tplg_filename	= "sof-acp.tplg",
```
Comment (plbossart) [inline sound/soc/sof/amd/pci-vangogh.c:59]: the same topology name is used for renoir, is this intentional?

File: sound/soc/sof/amd/acp-loader.c [hunk 10]
```diff
@@ -219,3 +223,34 @@ int acp_sof_dsp_run(struct snd_sof_dev *sdev)
...
+
+int acp_sof_load_signed_firmware(struct snd_sof_dev *sdev)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct acp_dev_data *adata = plat_data->hw_pdata;
+	int ret;
+
+	ret = request_firmware(&sdev->basefw.fw, adata->fw_code_bin, sdev->dev);
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: sof signed firmware code bin is missing\n");
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-loader.c]: we don't use the error: prefix any longer, please don't add it in new contributions

File: sound/soc/sof/amd/acp.h [hunk 11]
```diff
@@ -82,6 +83,8 @@
 #define SRAM1_SIZE				0x13A000
 #define PROBE_STATUS_BIT			BIT(31)
 
+#define FIRMWARE_SIGNATURE			0x100
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.h]: prefix?

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/sof/amd/pci-vangogh.c (hunk 9 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/pci-vangogh.c:59]: Yes,
We are using sof_pdata->[tplg_filename]
nocodec_tplg_filename is not using, so we mentioned default topology name for renior and this platform as well.

File: sound/soc/sof/amd/acp-loader.c (hunk 10 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp-loader.c]: Okay, we will remove error: prefix

File: sound/soc/sof/amd/acp.h (hunk 11 above)
Comment (Venkata-Prasad-Potturu) [reply sound/soc/sof/amd/acp.h]: Okay, Thanks : )
We will change it  to ACP_FIRMWARE_SIGNATURE.

**Review by plbossart — APPROVED** [review]

**Review by vijendarmukunda — APPROVED** [review]

**Comment by plbossart** [comment]
test this please

**Comment by plbossart** [comment]
SOFCI TEST

**Review by bhiregoudar — APPROVED** [review]

**Comment by vijendarmukunda** [comment]
@plbossart :We have to push some more fixes on top of current PR for AMD SOF stack. Could you please merge this PR changes?

**Pull request merged**